CHANGELOG.md

```diff
@@ -1,5 +1,53 @@
+* Support for all types of attribute values, not just strings.
+* All values are passed through to notification listeners.
+* Strings, booleans, and valid numbers are passed to the event dispatcher and can be used for Optimizely results segmentation. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³]. 
+* Strings, booleans, and valid numbers are relevant for audience conditions.
+* Support for additional matchers in audience conditions:
+* An `exists` matcher that passes if the user has a non-null value for the targeted user attribute and fails otherwise.
+* A `substring` matcher that resolves if the user has a string value for the targeted attribute.
+* `gt` (greater than) and `lt` (less than) matchers that resolve if the user has a valid number value for the targeted attribute. A valid number is a finite number in the inclusive range [-2⁵³, 2⁵³].
+* The original (`exact`) matcher can now be used to target booleans and valid numbers, not just strings.
+* Support for A/B tests, feature tests, and feature rollouts whose audiences are combined using `"and"` and `"not"` operators, not just the `"or"` operator.
+* Datafile-version compatibility check: The SDK will remain uninitialized (i.e., will gracefully fail to activate experiments and features) if given a datafile version greater than 4.
+* Updated Pull Request template and commit message guidelines.
+* Only remove old variation assignments from the user profile service when there are >= 100 persisted assignments.
+* Support for empty bucketing IDs when evaluating feature rollouts.
+
+### Breaking Changes:
+* Previously, notification listeners were only given string-valued user attributes because only strings could be passed into various method calls. That is no longer the case. The ActivateListener and TrackListener interfaces now receive user attributes as `NSDictionary<NSString *, NSObject *> * _Nullable` instead of  `NSDictionary<NSString *,NSString *> * _Nonnull`.
+* The Localytics integration now relies on the user profile service to attribute events to the user's variations.
+* Drops support for the `getVariableBoolean`, `getVariableDouble`, `getVariableInteger`, and `getVariableString` APIs. Please migrate to the Feature Management APIs which are available in all 2.x releases.
+
+### Bug Fixes:
+* Experiments and features can no longer activate when a negatively targeted attribute has a missing, null, or malformed value.
+. * Audience conditions (except for the new `exists` matcher) no longer resolve to `false` when they fail to find an legitimate value for the targeted user attribute. The result remains `null` (unknown). Therefore, an audience that negates such a condition (using the `"not"` operator) can no longer resolve to `true` unless there is an unrelated branch in the condition tree that itself resolves to `true`.
+* Support for empty user IDs.
+* Updated JSONModel definitions.
```

Comment:
@jaeopt @thomaszurkan-optimizely feel free to update this entry. There were a couple PRs related to this and I wasn't following what they were doing.